src/hooks/useEvm/useEvm.ts

```diff
@@ -1,16 +1,23 @@
-import { CHAIN_NAMESPACE, fromChainId } from '@shapeshiftoss/caip'
+import { CHAIN_NAMESPACE, ChainId, fromChainId } from '@shapeshiftoss/caip'
 import { ETHWallet } from '@shapeshiftoss/hdwallet-core'
-import { useEffect, useMemo, useState } from 'react'
+import { useCallback, useEffect, useMemo, useState } from 'react'
 import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
 import { useWallet } from 'hooks/useWallet/useWallet'
 import { bnOrZero } from 'lib/bignumber/bignumber'
 import { selectFeatureFlags } from 'state/slices/preferencesSlice/selectors'
 import { useAppSelector } from 'state/store'
 
+const chainIdFromEthNetwork = (ethNetwork: string | null, supportedEvmChainIds: ChainId[]) => {
```

Comment:
Added to avoid logic duplication

Reply:
we wanna slap an explicit return type on this or she gud?

Reply:
Implicit will give us all we need, but agree explicit is always best: 1d4a8e5f7419b6570635f08542d30db3a085b67c